proto/pulumi/resource.proto

```diff
@@ -42,6 +42,13 @@ service ResourceMonitor {
     // Registers a package and allocates a packageRef. The same package can be registered multiple times in Pulumi.
     // Multiple requests are idempotent and guaranteed to return the same result.
     rpc RegisterPackage(RegisterPackageRequest) returns (RegisterPackageResponse) {}
+
+    // WaitForShutdown blocks until the resource monitor is canceled, which will
+    // happen once all the steps have executed. This allows the language runtime
+    // to stay running and handle callback requests, even after the user program
+    // has completed. Runtime SDKs should call this after executing the user's
+    // program. This can only be called once.
```

Comment:
```suggestion
    // program. This can only be called once.
```
Why? Should be fine to call multiple times, we might need to call it multiple times for things like multi-stack runs.

Reply:
I think we want one call per program. If we run multiple programs in the future, the eval source needs to know how many "nil" events should be coming from the programs so it can stop producing events once it has received them all.

Reply:
Good call, I went with the wordy `SignalAndWaitForShutdown`. Happy to pick something shorter, but this is pretty clear I think.